src/client/theme-default/components/VPSocialLink.vue

```diff
@@ -1,29 +1,69 @@
 <script lang="ts" setup>
 import type { DefaultTheme } from 'vitepress/theme'
 import { computed } from 'vue'
-import { icons } from '../support/socialIcons'
 
 const props = defineProps<{
   icon: DefaultTheme.SocialLinkIcon
   link: string
   ariaLabel?: string
 }>()
 
+const svgSpriteRegex = __SOCIAL_SVG_SPRITE_ICONS__
+  ? /\.\/sprite-social-icons\/VP(.*)Icon\.vue/
+  : /\.\/social-icons\/VP(.*)Icon\.vue/
+
+const icons = Object.entries(__SOCIAL_SVG_SPRITE_ICONS__
+    ?
+    import.meta.glob('./sprite-social-icons/*.vue', {
```

Comment:
we can use `useData().site.value.themeConfig.socialLinks` to load only configured social icons

Reply:
in fact we can inject the list via define